src/Codeception/Lib/Driver/Facebook.php

```diff
@@ -30,83 +42,107 @@ public function __construct($config, $logCallback = null)
     {
+        $appToken = $this->appId . '|' . $this->appSecret;
         $response = $this->executeFacebookRequest(
-            $this->appSession,
             'POST',
-            '/' . FacebookSession::_getTargetAppId() . '/accounts/test-users',
+            $this->appId . '/accounts/test-users',
+            $appToken,
             [
                 'name' => $name,
                 'installed' => true,
                 'permissions' => $permissions
             ]
-        )->getRawResponse();
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
     }
 
     public function deleteTestUser($testUserID)
     {
+        $appToken = $this->appId . '|' . $this->appSecret;
         $this->executeFacebookRequest(
-            $this->appSession,
             'DELETE',
-            '/' . $testUserID
+            '/' . $testUserID,
+            $appToken
         );
     }
 
     public function getTestUserInfo($testUserAccessToken)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'GET',
-            '/me'
-        )->getRawResponse();
+            '/me',
+            $testUserAccessToken
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
     }
 
     public function getLastPostsForTestUser($testUserAccessToken)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'GET',
-            '/me/feed'
-        )->getRawResponse();
+            '/me/feed',
+            $testUserAccessToken
+        );
 
-        return json_decode($response, true);
+        return $response->getDecodedBody();
+    }
+
+    public function getVisitedPlaceTagForTestUser($placeId, $testUserAccessToken)
+    {
+        $response = $this->executeFacebookRequest(
+            'GET',
+            "/$placeId",
+            $testUserAccessToken
+        );
+
+        return $response->getDecodedBody();
     }
 
     public function sendPostToFacebook($testUserAccessToken, array $parameters)
     {
         $response = $this->executeFacebookRequest(
-            new FacebookSession($testUserAccessToken),
             'POST',
             '/me/feed',
+            $testUserAccessToken,
             $parameters
-        )->getRawResponse();
-        return json_decode($response, true);
+        );
+        return $response->getDecodedBody();
     }
 
+
     /**
-     * @param FacebookSession $session
      * @param string $method
      * @param string $endpoint
      * @param array $parameters
+     * @param string $token
      * @return \Facebook\FacebookResponse
      */
-    private function executeFacebookRequest(FacebookSession $session, $method, $endpoint, array $parameters = [])
+    private function executeFacebookRequest($method, $endpoint, $token = null, array $parameters = [])
```

Comment:
IMO, optional `$token` is a potential issue; according to the FacebookSDK code, if `$accessToken` is null, then it will use `$defaultAccessToken`, which is not being set, so my suggestion here would be to use app token, if `$token` is null;  additionaly we can swap `$token` and `$parameters`, so that methods `createTestUser()` and `deleteTestUser()` can be simplified. what do you think ?